### PR TITLE
fix(DB/Quest): Rework 'A Spirit Guide' (9410)

### PR DESCRIPTION
**Pull request by LeFroid**
Fixes behavior of ancestral spirit wolf

## Changes Proposed:
-  Add waypoint to spirit wolf
-  Player can now follow wolf to Mag'har camp, instead of the wolf following the player wherever they go

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/14594

## Tests Performed:

- Tested quest to completion

## How to Test the Changes:

1. .q add 9410
2. .go c id 17062
3. Use item given in step 1 to summon spirit wolf (item 23669)
4. Follow wolf to Gorkan Bloodfist (16845)

## Known Issues and TODO List:

- The aura sets Mag'har faction to friendly when within range of the spirit wolf, but if the player never leaves the radius of the wolf from the moment it's summoned, the faction change is not applied. That is to say, the player has to leave the range of the aura, and walk back into range for it to apply properly. I am not sure why this is.
- The NPC Earthcaller Ryga has a scripted event, not necessary for quest completion, which is not working. When the wolf finishes the waypoint, it triggers the smart script by setting the data field 0 of Ryga (id 17123) to value 1 (uses combination of `SMART_EVENT_DATA_SET` and `SMART_EVENT_SET_DATA`). Even when manually running `.npc set data 0 1` which should trigger the same, nothing happens. Perhaps this is a larger issue with smart script type `SMART_EVENT_DATA_SET`
- After testing this change a second time, the wolf was despawned right before reaching the final waypoint (did not happen on first test). This makes the change **unacceptable** at the moment.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Gultask** [comment]
What's the source of the waypoints?

**Review by Gultask — COMMENTED** [review]
Use single quotes instead of double quotes

File: data/sql/updates/pending_db_world/rev_1680983241816980712.sql [hunk 1]
```diff
@@ -0,0 +1,134 @@
+-- Ancestral Spirit Wolf SAI (Source: Sniffs & https://www.youtube.com/watch?v=xFF8DOSeMcg)
+SET @ID := 17077;
+UPDATE `creature_template` SET `speed_walk` = 1.6, `speed_run` = 1, `unit_flags` = 768, `AIName` = "SmartAI", `ScriptName` = "" WHERE `entry` = @ID;
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/rev_1680983241816980712.sql:3]: Speeds are already sniffed: https://github.com/neifion-00000000/azerothcore-wotlk/blob/76e9f511a1d3f08f3fe9d224c7ae488c39930a5c/data/sql/updates/pending_db_world/rev_1672823650132911600.sql
```suggestion
UPDATE `creature_template` SET `unit_flags` = 768, `AIName` = "SmartAI", `ScriptName` = "" WHERE `entry` = @ID;
```

**Comment by Gultask** [comment]
Thanks for the PR!
Is this a cherry-pick?

**Comment by acidmanifesto** [comment]
> Thanks for the PR! Is this a cherry-pick?

https://github.com/TrinityCore/TrinityCore/commit/c905af818c81d41d5e0b025de97c0cdbd1b417cb#diff-236b08d19a9b383633264786da628ee1210eafdee55addce2e318ba526e48a7c

**Comment by LeFroid** [comment]
Thanks @acidmanifesto 
I will make your proposed changes @Gultask . In the meantime, I tested this change a second time and found the spirit wolf to despawn right before reaching the final waypoint. This made the quest incompletable. I will have to look further into root cause

**Comment by pklloveyou** [comment]
image
I fixed them through hacker code

File: data/sql/updates/pending_db_world/rev_1680983241816980712.sql (hunk 1 above)
Comment (LeFroid) [reply data/sql/updates/pending_db_world/rev_1680983241816980712.sql]: Updated accordingly, but still facing the issue where the script is behaving inconsistently. Can we add a WIP label to the PR? I don't know how or don't have permission to do so.

**Comment by LeFroid** [comment]
Closing this PR

**Pull request closed without merging**